### PR TITLE
feat: mempool rescan

### PR DESCRIPTION
**Pull request by michael1011**
Todos:
- [x] fetching relevant inputs and outputs for the mempool rescan from database
- [x] tests

### CODE REVIEW

**Review by maybeast — APPROVED** [review]

**Review by maybeast — CHANGES_REQUESTED** [review]
giving it another look

File: boltzr/src/swap/filters.rs [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+
+pub type Filters = HashMap<String, (HashSet<Outpoint>, HashSet<Vec<u8>>)>;
+
+pub fn get_chain_filters(
+    currencies: &Currencies,
+    swap_repo: &Arc<dyn SwapHelper + Sync + Send>,
+    reverse_swap_repo: &Arc<dyn ReverseSwapHelper + Sync + Send>,
+    chain_swap_repo: &Arc<dyn ChainSwapHelper + Sync + Send>,
+) -> Result<Filters> {
+    let pending_swaps = swap_repo.get_all(Box::new(
```
Comment (maybeast) [inline boltzr/src/swap/filters.rs:24]: Consider breaking this function down to something like `process_pending_swaps`, `process_pending_reverse_swaps` and `process_pending_chain_swaps`

**Review by maybeast — CHANGES_REQUESTED** [review]

File: boltzr/src/chain/utils.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        match self {
+            Transaction::Bitcoin(tx) => tx
+                .output
+                .iter()
+                .map(|o| o.script_pubkey.to_bytes())
+                .collect(),
+            Transaction::Elements(tx) => tx
+                .output
+                .iter()
+                .filter(|o| !o.script_pubkey.is_empty())
```
[the remark this answers is not in the document]
Comment (michael1011) [reply boltzr/src/chain/utils.rs:57]: In Elements an empty script is used for the explicit fee output; in Bitcoin empty scripts are not allowed

File: boltzr/src/db/helpers/swap.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+use crate::db::helpers::{BoxedCondition, QueryResponse};
+use crate::db::models::Swap;
+use crate::db::schema::swaps;
+use crate::db::Pool;
+use diesel::{QueryDsl, RunQueryDsl, SelectableHelper};
+
+pub trait SwapHelper {
+    fn get_all(&self, condition: BoxedCondition<swaps::dsl::swaps>) -> QueryResponse<Vec<Swap>>;
```
[the remark this answers is not in the document]
Comment (michael1011) [reply boltzr/src/db/helpers/swap.rs:8]: `::table` and `sdsl::swaps` are aliases

**Review by maybeast — APPROVED** [review]

**Pull request merged**